controllers/slice/reconciler.go

```diff
@@ -69,7 +69,7 @@ type SliceReconciler struct {
 //+kubebuilder:rbac:groups=networking.istio.io,resources=gateways,verbs=get;list;create;update;watch;delete
 //+kubebuilder:rbac:groups=networking.istio.io,resources=serviceentries,verbs=get;list;create;update;watch;delete
 //+kubebuilder:rbac:groups=networking.istio.io,resources=virtualservices,verbs=get;list;create;update;watch;delete
-//+kubebuilder:webhook:path=/mutate-appsv1-deploy,mutating=true,failurePolicy=fail,groups="apps",resources=deployments,verbs=create;update,versions=v1,name=mdeploy.kubeslice.io,admissionReviewVersions=v1,sideEffects=NoneOnDryRun
+//+kubebuilder:webhook:path=/mutate-appsv1-deploy,mutating=true,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=mdeploy.kubeslice.io,admissionReviewVersions=v1,sideEffects=NoneOnDryRun
```

Comment:
should we change the service path and webhook name as well?  it looks a bit weird to call the webook `mdeploy` when it is actually mutating pod instead of deployments.
This will require helm chart changes though

Reply:
helm chart has to be changed anyway. so i would say lets fix the names as well

Reply:
can we call it webhook.kubeslice.io

Reply:
yeah that sounds good @Rahul-D78